src/linalg/Gesvd_truncate.cpp

```diff
@@ -381,6 +383,248 @@ namespace cytnx {
 
     }  // Gesvd_truncate
 
+    void _gesvd_truncate_Block_UT(std::vector<UniTensor> &outCyT, const cytnx::UniTensor &Tin,
+                                  const cytnx_uint64 &keepdim,
+                                  std::vector<cytnx_uint64> min_blockdim, const double &err,
+                                  const bool &is_U, const bool &is_vT,
+                                  const unsigned int &return_err, const cytnx_uint64 &mindim) {
+      // currently, Gesvd is used as a standard for the full SVD before truncation
+      cytnx_int64 keep_dim = keepdim;  // these must be signed int, because they can become
+                                       // negative!
+      cytnx_int64 min_dim = (mindim < 1 ? 1 : mindim);
+
+      outCyT = linalg::Gesvd(Tin, is_U, is_vT);
+      if (min_blockdim.size() == 1)  // if only one element given, make it a vector
+        min_blockdim.resize(outCyT[0].Nblocks(), min_blockdim.front());
+      cytnx_error_msg(
+        min_blockdim.size() != outCyT[0].Nblocks(),
+        "[ERROR][Gesvd_truncate] min_blockdim must have the same number of elements as "
+        "blocks in the singular value UniTensor%s",
+        "\n");
+
+      // process truncation:
+      // 1) concate all S vals from all blk but exclude the first min_blockdim Svals in each block
+      // (since they will be kept anyways later)
+      Tensor Sall;  // S vals excluding the already kept ones
+      Tensor Block;  // current block
+      cytnx_uint64 blockdim;
+      bool anySall = false;  // are there already any values in Sall vals?
+      bool any_min_blockdim = false;  // is any min_blockdim > 0?
+      for (int b = 0; b < outCyT[0].Nblocks(); b++) {
+        if (min_blockdim[b] < 1)  // save whole block to Sall
+          Block = outCyT[0].get_block_(b);
+        else {
+          any_min_blockdim = true;
+          blockdim = outCyT[0].get_block_(b).shape()[0];
+          if (blockdim <= min_blockdim[b]) {
+            // keep whole block
+            keep_dim -= blockdim;
+            min_dim -= blockdim;
+            continue;
+          }
+          // remove first min_blockdim[b] values
+          blockdim = outCyT[0].get_block_(b).shape()[0];
+          Block = outCyT[0].get_block_(b).get({ac::range(min_blockdim[b], blockdim)});
```

Comment:
Does this line remove the largest `min_blockdim[b]` numbers?

Reply:
@manuschneider Is this intended?

Reply:
It is intended. The first `min_blockdim[b]` singular values will be saved later anyways. So they should not be included in `Sall`. The latter only contains the remaining singular values that need to be ordered to select which ones should be kept.